### DISCUSSION TITLE
How to access Live Server preview from another device on LAN

### DISCUSSION THREAD

**Discussion by tophneal** [coder/code-server#4360] [Q&A] [answered]
I'm honestly not sure if this would be considered an issue I'm having with code-server or with Docker. Hopefully someone can point me in the right direction.

I've got a server I use to run docker instances/services with. I recently installed code-server in it's own container, setup HTTPS with it to appease the service workers, and can now get live-server and five-server extensions to run. I love the idea of having the editor available on any computer with a browser and access to my network through a VPN.

However, after starting the extensions, regardless of the extension settings, I cannot get access to the preview. In the browser window that comes up the address is always shown as 127.0.0.1. If i change it to the docker server's IP, where i have the port for extension opened/forwarded I just get a 404. I've tried changing the settings to use 0.0.0.0 and 127.0.0.1 i still used, and I've tried setting it to use the Docker server's IP, and i get a refusal to connect. In the Terminal pane I can see "Five Server running at: > Local: http://localhost:5555 >Network: http://172.17.0.4:555"

Thanks!

**Comment by jsjoeio** [comment, marked as answer]
> I cannot get access to the preview

What do you mean by "preview"? Is this a Markdown preview or accessing a web service or something else?

> I've tried changing the settings to use 0.0.0.0 and 127.0.0.1 i still used, and I've tried setting it to use the Docker server's IP, and i get a refusal to connect

Hmm... @code-asher any ideas?

**Reply by tophneal** [reply]
My bad. I’m trying to access a live preview web service, for testing CSS/JavaScript. The extension is just a mini web server with a browser pane. 

sorry about any typos or weird wording in my OP, Apples autocorrect can be aggressively wrong, and I don’t always catch it.

**Reply by lg-nineoneninethreenine** [reply]
I know Live Server. It kinda activates a server on port 5555, is a VS Code extension by Ritwick Dey and is also available via Microsoft. I think if you allow external connections to the Docker deployment or use ngrok's HTTP port forwarding, you can view it. Do note that the URL changes every time, so you have to copy a new link.

**Reply by lg-nineoneninethreenine** [reply]
On LAN, idk, but you can view it like any website via ngrok.

**Reply by code-asher** [reply]
Yeah I have not tried this but it sounds like you will need to proxy it somehow, whether via something like ngrok or code-server's built-in proxy.  So you would have Live Server listening on `localhost:5555` then in the preview it would need to use `{code-server host or ip}/proxy/5555` (or `{code-server host or ip}/absproxy/5555` if it needs a non-rewriting proxy).

**Reply by code-asher** [reply]
But exposing 5555 through the same mechanism you expose code-server should also work.

**Reply by jsjoeio** [reply]
Let us know if that works!

**Comment by tophneal** [comment]
Thanks! I’ll look into that.

**Reply by tophneal** [reply]
Well I'm dumbstruck now. Five (Live) Server just started working! I already had port 5555 forwarded in the container, but just now, for S's & G's, I retried opening an HTML file with it. It opened the browser window to a 127.0.0.1 address, but this time I was able to replace it with my Docker server's local IP and view the page! Woohoo! I will likely look into ngrok, to see if I can get it to point to the correct IP when opening, though.

Feel free to close this ticket, if you'd like!

**Reply by jsjoeio** [reply]
Interesting! Well, glad to hear that it is working now! Thanks for letting us know!
